lib/wallet/test/unit/Cardano/Wallet/Submissions/Generation.hs

```diff
@@ -0,0 +1,155 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE NoMonoLocalBinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.Wallet.Submissions.Generation
+    ( prop_submission_history
+    , SubmissionsHG (..)
+    , txG
+    , slotG
+    , noSubmissions
+    , Step (..)
+    , P
+    )
+where
+
+import Prelude
+
+import Cardano.Wallet.Submissions.Properties.Common
+    ( Step (..), newState )
+import Cardano.Wallet.Submissions.Submissions
+    ( Submissions (Submissions), transactionsL )
+import Cardano.Wallet.Submissions.TxStatus
+    ( HasTxId (..), getTx, _Expired, _InLedger, _InSubmission )
+import Control.Arrow
+    ( (&&&) )
+import Control.Lens
+    ( lastOf, (&), _2 )
+import Control.Lens.Extras
+    ( is )
+import System.Random
+    ( Random )
+import Test.QuickCheck
+    ( Arbitrary (arbitrary), Gen, choose, elements, frequency, getSize )
+import Test.QuickCheck.Property
+    ( Property, conjoin, cover, forAllShrinkShow, mapSize )
+
+import Data.Foldable
+    ( toList )
+import qualified Data.Map.Strict as Map
+import Data.Maybe
+    ( fromJust )
+import qualified Data.Text.Lazy as T
+import Text.Pretty.Simple
+    ( pShow )
+
+newtype Slot = Slot Int
+    deriving (Eq, Show, Ord, Random, Enum, Num, Arbitrary)
+
+newtype Tx = Tx Int
+    deriving (Eq, Show, Ord, Arbitrary)
+
+type P x = x Slot Tx
```

Comment:
Could you add a documentation comment what `P` means? (I appreciate its utility.)